test/extended/deployments/deployments.go

```diff
@@ -185,6 +186,29 @@ var _ = g.Describe("deploymentconfigs", func() {
 			}
 		})
 	})
+
+	g.Describe("with custom deployments", func() {
+		g.It("should run the custom deployment steps [Conformance]", func() {
+			out, err := oc.Run("create").Args("-f", customDeploymentFixture).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			o.Expect(waitForLatestCondition(oc, "custom-deployment", deploymentRunTimeout, deploymentRunning)).NotTo(o.HaveOccurred())
+
+			out, err = oc.Run("logs").Args("-f", "dc/custom-deployment").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By(fmt.Sprintf("checking the logs for substrings\n%s", out))
+			o.Expect(out).To(o.ContainSubstring("--> pre: Running hook pod ..."))
+			o.Expect(out).To(o.ContainSubstring("test pre hook executed"))
+			o.Expect(out).To(o.ContainSubstring("--> Scaling custom-deployment-1 to 2"))
```

Comment:
yes, the % that is input is "how far in the new deployment", not "how far
in the total deployment", and is best effort (if you ask for 50%, and we go
to 100%, we don't stop).  If you want 50% I think it would be expected you
would set your surge / maxUnavailable to achieve that.

On Tue, May 10, 2016 at 12:32 PM, Michail Kargakis <notifications@github.com
> wrote:

> In test/extended/deployments/deployments.go
> <https://github.com/openshift/origin/pull/8787#discussion_r62705648>:
>
> > @@ -185,6 +186,29 @@ var _ = g.Describe("deploymentconfigs", func() {
> >  			}
> >  		})
> >  	})
> > +
> > +	g.Describe("with custom deployments", func() {
> > +		g.It("should run the custom deployment steps [Conformance]", func() {
> > +			out, err := oc.Run("create").Args("-f", customDeploymentFixture).Output()
> > +			o.Expect(err).NotTo(o.HaveOccurred())
> > +
> > +			o.Expect(waitForLatestCondition(oc, "custom-deployment", deploymentRunTimeout, deploymentRunning)).NotTo(o.HaveOccurred())
> > +
> > +			out, err = oc.Run("logs").Args("-f", "dc/custom-deployment").Output()
> > +			o.Expect(err).NotTo(o.HaveOccurred())
> > +			g.By(fmt.Sprintf("checking the logs for substrings\n%s", out))
> > +			o.Expect(out).To(o.ContainSubstring("--> pre: Running hook pod ..."))
> > +			o.Expect(out).To(o.ContainSubstring("test pre hook executed"))
> > +			o.Expect(out).To(o.ContainSubstring("--> Scaling custom-deployment-1 to 2"))
>
> If you scale to 2 you scale to 100%, right?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8787/files/a96fb629083f11e94f9496eb3fd5d5fa15ab9245#r62705648>
>